src/app/drive/components/StopSharingItemDialog/StopSharingItemDialog.tsx

```diff
@@ -0,0 +1,43 @@
+import { useTranslationContext } from '../../../i18n/provider/TranslationProvider';
+import Button from '../../../shared/components/Button/Button';
+import Modal from '../../../shared/components/Modal';
+import Spinner from '../../../shared/components/Spinner/Spinner';
+
+const StopSharingItemDialog = ({
+  showStopSharingConfirmation,
+  onClose,
+  name,
+  isLoading,
+  onStopSharing,
+}: {
+  showStopSharingConfirmation: boolean;
+  onClose: () => void;
+  name: string;
+  isLoading: boolean;
+  onStopSharing: () => void;
+}) => {
+  const { translate } = useTranslationContext();
+  return (
+    <Modal
+      maxWidth="max-w-sm"
+      className="space-y-5 p-5"
+      isOpen={showStopSharingConfirmation}
+      onClose={() => onClose()}
+      preventClosing={showStopSharingConfirmation && isLoading}
+    >
+      <p className="text-2xl font-medium">{translate('modals.shareModal.stopSharing.title')}</p>
+      <p className="text-lg text-gray-80">{translate('modals.shareModal.stopSharing.subtitle', { name: name })}</p>
+      <div className="flex items-center justify-end space-x-2">
+        <Button variant="secondary" onClick={() => onClose()} disabled={showStopSharingConfirmation && isLoading}>
```

Comment:
you can pass the reference of the function `onClose` instead of creating an anonymous one and executing it, that way it will not be destroyed and will create a new one with each rerender.